modules/core/04-channel/v2/keeper/packet.go

```diff
@@ -173,6 +162,62 @@ func (k Keeper) recvPacket(
 	return nil
 }
 
+func (k Keeper) acknowledgePacket(ctx context.Context, packet channeltypesv2.Packet, acknowledgement channeltypesv2.Acknowledgement, proof []byte, proofHeight exported.Height) error {
+	// Lookup counterparty associated with our channel and ensure
+	// that the packet was indeed sent by our counterparty.
+	counterparty, ok := k.GetCounterparty(ctx, packet.SourceId)
```

Comment:
this will become a v2 channelEnd, and below `clientID := counterparty.ClientID` will read better... right now its slightly confusing. 

The channelEnd.ClientID would suggest to me that it's the clientID of the client representing the counterparty state machine. Rather than a counterparty.ClientID which would suggest to me its the clientID on the counterparty which repesents _this_ chain (self host where this code is being executed)

Reply:
yeah the naming is a bit all over the place at the moment with regards to counterparty/channel/ids etc, hopefully we'll be able to sort this out quite soon!

Reply:
would be nice to get this rename done and dusted sooner rather than later methinks!